### ISSUE TITLE
Transformers are broken since 2.0.0-beta.10

### ISSUE THREAD

**Issue by urmuzov** [cloudendpoints/endpoints-java#86]
@tangiel we encountered a problem when tried to upgrade our endpoints version from 2.0.0-beta.8 to 2.0.6

The problem appeared with this commit https://github.com/cloudendpoints/endpoints-java/commit/0b42dd6748d3703cd4fad1d58df7a8ce0bd9d16b 

If you have transformer like:
```java
public final class UuidTransformer implements Transformer<UUID, String> {
    @Override
    public String transformTo(final UUID key) {
        return key.toString();
    }

    @Override
    public UUID transformFrom(final String s) {
        return UUID.fromString(s);
    }
}
```

and API method like:
```java
    @ApiMethod(name = "echo_uuid", path = "echo_uuid", httpMethod = ApiMethod.HttpMethod.GET)
    public Message echoUuid(@Named("uuid") UUID uuid, @Named("n") @Nullable Integer n) {
        ...
    }
```

framework is not able to generate discovery doc.

We made example for you to test this bug: [backend.zip](https://github.com/cloudendpoints/endpoints-java/files/930012/backend.zip)

How to reproduce the problem:

- Set version in pom.xml `<endpoints.framework.version>2.0.0-beta.9</endpoints.framework.version>`
- `mvn clean install appengine:run`
- Go to `http://localhost:8080/_ah/api/explorer` - it works

- Set version in pom.xml `<endpoints.framework.version>2.0.0-beta.10</endpoints.framework.version>`
- `mvn clean install appengine:run`
- Go to `http://localhost:8080/_ah/api/explorer` - it fails with
```
[INFO] GCLOUD: SEVERE: Could not generate or cache discovery doc
[INFO] GCLOUD: java.lang.IllegalArgumentException: Can't add a primitive type as a resource
```

**Comment by tangiel** [comment]
Seems the issue is not having a named `UUID`, but having `UUID` as a field, as demonstrated by the example. I believe I know what's going on and should have a fix coming up.

**Comment by urmuzov** [comment]
It's the same thing with `org.bson.types.ObjectId` or our own types that transformed to and from `java.lang.String`.

In our example the main point is the Message class that contains `List<UUID>`:

```java
public class Message {

    private String message;
    private List<UUID> to;

...
}
```

And looks like it's all collection types. We have the same problem with `Set<ObjectId>`. Not sure about maps.

Does this makes sense?

**Comment by tangiel** [comment]
Fix is in #89 and will be released in 2.0.7.

**Comment by urmuzov** [comment]
@tangiel Just checked 2.0.7
Looks like you've fixed it :) Thanks!
